motionplan/tpSpaceRRT.go

```diff
@@ -988,14 +939,15 @@ func (mp *tpSpaceRRTMotionPlanner) attemptSmooth(
 
 	reachedDelta := mp.planOpts.DistanceFunc(&ik.Segment{StartPosition: reached.Pose(), EndPosition: path[secondEdge].Pose()})
 	// If we tried the goal and have a close-enough XY location, check if the node is good enough to be a final goal
-	if reachedDelta > mp.algOpts.poseSolveDist {
+	if reachedDelta > mp.planOpts.GoalThreshold {
 		return nil, errors.New("could not precisely reach smoothing destination")
 	}
-
 	newInputSteps := extractPath(startMap, nil, &nodePair{a: reached.node, b: nil}, false)
 
 	if secondEdge < len(path)-1 {
 		newInputSteps = append(newInputSteps, path[secondEdge+1:]...)
+	} else {
```

Comment:
I dont understand this case

Reply:
If `secondEdge` is the last node of the plan, then it's the node at the goal pose whose configuration should be `0, 0, 0`.

`newInputSteps` will not contain this `0, 0, 0` node because it just extended to it. But `path[secondEdge+1:]` will not include it either, it will reach past the end of the path.

So basically this is saying "if we smoothed all the way to the goal node, then that smoothing process will have removed the path endpoint node, so replace it".

Reply:
Can you comment something like this?  This is the type of code that would be very hard to understand without you here to explain it

Reply:
Can you comment something like this? This is the type of code that would be very hard to understand without you here to explain it

Reply:
Done!